docs/diagrams/models/ResidentRoomZoomIn.puml

```diff
@@ -0,0 +1,47 @@
+@startuml
+!include ../style.puml
+skinparam arrowThickness 1.1
+skinparam arrowColor MODEL_COLOR
+skinparam classBackgroundColor MODEL_COLOR
+skinparam linetype ortho
+
+Package Model <<Rectangle>>{
+    Class StatefulAddressBook
+    Class ModelManager
+    Package ResidentRoom {
+        Class UniqueResidentRoomList
+        Class ResidentRoom
+    }
+    Package Resident {
+        Class UniqueResidentList
+        Class Resident
+        Class Name
+    }
+    Package Room {
+        Class UniqueRoomList
+        Class Room
+        Class RoomNumber
+    }
+}
+' ===== Overarching stuff =====
+ModelManager "1" *-> "1" StatefulAddressBook
+
+' ==== Room ====
+ModelManager  *---> "~* filteredlist" Room
+StatefulAddressBook --down-> "1" UniqueRoomList
```

Comment:
As mentioned in another comment, I think it is the `AddressBook` that references the sub-components. This also affects `IssueZoomIn.puml`, `AliasZoomIn.puml` and `CommandHistoryZoomIn.puml`